internal/app/connectconformance/results.go

```diff
@@ -459,7 +459,7 @@ func checkPayloads(expected, actual []*conformancev1.ConformancePayload) multiEr
 	return errs
 }
 
-func checkError(expected, actual *conformancev1.Error) multiErrors {
+func checkError(expected, actual *conformancev1.Error, otherCodes []conformancev1.Code) multiErrors {
```

Comment:
Would it make sense to use a `map[string]struct{}` or other `Set` type here to make searching easier?

Reply:
The list of codes is always small, so linear scan seemed fine (likely faster than having to create a map). As far as just simplifying the code (since perf is not a big concern here), I think constructing the map from the repeated proto field would be the same amount of code as the linear search through the slice, so didn't seem like a clear win. Also, the linear search is part of the standard library in Go 1.21, which introduces a new `slices` package, so we could upgrade in the near future and then remove some of this code.

I can add a comment to the `inSlice` function about removing it once we update to Go 1.21.